ds3-sdk-integration/src/test/java/com/spectralogic/ds3client/integration/PutJobManagement_Test.java

```diff
@@ -0,0 +1,456 @@
+                    wrap(client).startWriteJob(bucketName, Lists.newArrayList(new Ds3Object("testOne", 2)));
+            final GetJobSpectraS3Response jobResponse = client
+                    .getJobSpectraS3(new GetJobSpectraS3Request(job.getJobId()));
+
+            final Date originalDate = jobResponse.getMasterObjectListResult().getStartDate();
+            final Date newDate = new Date(originalDate.getTime() - 1000);
+
+            client.modifyJobSpectraS3(new ModifyJobSpectraS3Request(job.getJobId())
+                    .withCreatedAt(newDate));
+
+            final GetJobSpectraS3Response responseAfterModify = client
+                    .getJobSpectraS3(new GetJobSpectraS3Request(job.getJobId()));
+
+            assertThat(responseAfterModify.getMasterObjectListResult().getStartDate(), is(newDate));
+
+        } finally {
+            deleteAllContents(client, bucketName);
+        }
+    }
+
+    @Test
+    public void cancelJob() throws IOException, SignatureException, XmlProcessingException {
+        final String bucketName = "test_cancel_job";
+        try {
+            Ds3ClientHelpers.wrap(client).ensureBucketExists(bucketName);
+
+            final Ds3ClientHelpers.Job job =
+                    wrap(client).startWriteJob(bucketName, Lists.newArrayList(new Ds3Object("testOne", 2)));
+
+            final CancelJobSpectraS3Response response = client
+                    .cancelJobSpectraS3(new CancelJobSpectraS3Request(job.getJobId()));
+            assertEquals(response.getStatusCode(),204);
+
+            assertTrue(client.getActiveJobsSpectraS3(new GetActiveJobsSpectraS3Request())
+                    .getActiveJobListResult().getActiveJobs().isEmpty());
+
+        } finally {
+            deleteAllContents(client, bucketName);
+        }
+    }
+
+    @Test
+    public void truncateJobCancelWithOutForce() throws IOException, SignatureException, XmlProcessingException, URISyntaxException, InterruptedException {
+
+        final int testTimeOutSeconds = 5;
+        final String bucketName = "test_truncate_cancel_job";
+        final String book1 = "beowulf.txt";
+        final Path objPath1 = ResourceUtils.loadFileResource(RESOURCE_BASE_NAME + book1);
+        final Ds3Object obj1 = new Ds3Object(book1, Files.size(objPath1));
+        final Ds3Object obj2 = new Ds3Object("place_holder", 5000000);
+
+        try {
+            Ds3ClientHelpers.wrap(client).ensureBucketExists(bucketName);
+
+            final Ds3ClientHelpers.Job putJob = Ds3ClientHelpers.wrap(client).startWriteJob(bucketName, Lists.newArrayList(obj1, obj2));
+            final UUID jobId = putJob.getJobId();
+            final SeekableByteChannel book1Channel = new ResourceObjectPutter(RESOURCE_BASE_NAME).buildChannel(book1);
+            client.putObject(new PutObjectRequest(bucketName, book1, book1Channel, jobId, 0, Files.size(objPath1)));
+
+            //make sure black pearl has updated it's job to show 1 object in cache
```

Comment:
This block of code is reused in multiple places.  It should be consolidated into a single function all and reused.